cpp/time_.h

```diff
@@ -9,21 +9,54 @@
 
 namespace time_ {
 
-inline void* tzset() {
-  NotImplemented();
+inline void tzset() {
+  ::tzset();
 }
 
-inline int time() {
+inline time_t time() {
   return ::time(nullptr);
 }
 
-// TODO: Should these be bigger integers?
-inline int localtime(int ts) {
-  NotImplemented();
+// NOTE(Jesse): time_t is specified to be an arithmetic type by C++. On most
+// systems it's a 64-bit integer.  64 bits is used because 32 will overflow in
+// 2038.  Someone on a comittee somewhere thought of that when moving to 64-bit
+// architectures to prevent breaking ABI again; on 32-bit systems it's usually
+// 32 bits.  Point being, using anything but the time_t typedef here could
+// (unlikely, but possible) produce weird behavior.
+inline time_t localtime(time_t ts) {
+  tm* loc_time = ::localtime(&ts);
+  time_t result = mktime(loc_time);
+  return result;
 }
 
-inline Str* strftime(Str* s, int ts) {
-  NotImplemented();
+inline Str* strftime(Str* s, time_t ts) {
+  Str* result = kEmptyString;
+
+  assert(s->IsNulTerminated());
+
+  tm* loc_time = ::localtime(&ts);
+
+  const int buf_size = 1024;
```

Comment:
Hm this isn't always enough?

Reply:
Hm I was going to say that returning `""` isn't the right behavior when the strftime format string  is big, because it is exposed to the user through the `printf` builtin -- see `osh/builtin_printf`. 

So if they pass a long string then it looks like it will silently go to `""`.

But then I tested out bash and that's exactly what it does.

https://github.com/oilshell/oil/commit/737d716d91813a0d04cd5120b3ed89960fde4f87

I still kind of prefer a user visible error though.  Maybe I will merge this and see if we can use `e_die()`